src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -34,6 +34,7 @@ export interface ITableFilterOptions {
 	 * The message to be displayed when the filter is disabled and the user tries to open the filter menu.
 	 */
 	disabledFilterMessage?: string;
+	shouldSetColumns?: boolean
```

Comment:
based on it's purpose, I'd rename it to refreshColumns, and add comment to explain this is to avoid redraw the grid multiple times. default value is true.